src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let currentState = Object.assign({}, state);
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      currentState = {};
+      history.push(currentState);
+    }
+
+    if (actions[i].type === 'addProperties') {
+      const add = actions[i].extraData;
+
+      currentState = Object.assign({}, currentState, add);
+      history.push(currentState);
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      currentState = Object.assign({}, currentState);
+
+      actions[i].keysToRemove.forEach((key) => {
+        delete currentState[key];
+      });
+      history.push(currentState);
+    }
+  }
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. This will make your code cleaner and more maintainable. You can refactor the if statements to a switch statement to handle the different action types.